src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20210702101641_EES2156_MigrateMethodologyTitleAndSlugToParentAndAddAlternativeTitleField.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Migrations.ContentMigrations
+{
+    public partial class EES2156_MigrateMethodologyTitleAndSlugToParentAndAddAlternativeTitleField : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<string>(
+                name: "OwningPublicationTitle",
+                table: "MethodologyParents",
+                nullable: false,
+                defaultValue: "");
+
+            migrationBuilder.AddColumn<string>(
+                name: "Slug",
+                table: "MethodologyParents",
+                nullable: false,
+                defaultValue: "");
+
+            migrationBuilder.AddColumn<string>(
+                name: "AlternativeTitle",
+                table: "Methodologies",
+                nullable: true);
+
+            // Set the default Methodology titles to fall back to the owning Publications' titles.
+            migrationBuilder.Sql(@"
+                UPDATE MethodologyParents 
+                SET MethodologyParents.OwningPublicationTitle = Publications.Title
+                FROM MethodologyParents 
+                JOIN PublicationMethodologies 
+                  ON PublicationMethodologies.MethodologyParentId = MethodologyParents.Id 
+                  AND PublicationMethodologies.Owner = 1 
+                JOIN Publications
+                  ON Publications.Id = PublicationMethodologies.PublicationId");
+            
+            // Explicitly set an alternative title for any Methodologies whose titles don't conform with those of their
+            // owning Publications.
+            migrationBuilder.Sql(@"
+                UPDATE Methodologies
+                SET Methodologies.AlternativeTitle = Methodologies.Title
+                FROM Methodologies
+                JOIN MethodologyParents ON MethodologyParents.Id = Methodologies.MethodologyParentId
+                WHERE Methodologies.Title != MethodologyParents.OwningPublicationTitle");
```

Comment:
Haha sounds good!  I will amend this slightly to be more forgiving :)

Reply:
Updated so that we set AlternaitveTitle if the Methodology Title doesn't match the Publiction Title exactly and if it doesn't match "{Publication title}: methodology".  So looking at the new update statement:

```
UPDATE Methodologies
                SET Methodologies.AlternativeTitle = Methodologies.Title
                FROM Methodologies
                JOIN MethodologyParents ON MethodologyParents.Id = Methodologies.MethodologyParentId
                WHERE Methodologies.Title != MethodologyParents.OwningPublicationTitle
                AND Methodologies.Title != CONCAT(MethodologyParents.OwningPublicationTitle, ': methodology')
```

If the Methodology title is "Hello" and the owning Publication title is "Pupil Absence", the first WHERE clause will return true and the second clause will return true, and so the AlternativeTitle will be set.

If the Methodology title is "Pupil Absence" and the owning Publication title is "Pupil Absence", the first WHERE clause will return false and so the AlternativeTitle will *not* be set.

If the Methodology title is "Pupil Absence: methodology" and the owning Publication title is "Pupil Absence", the first WHERE clause will return true but the second will return false, and so the AlternativeTitle will *not* be set.

Reply:
That's really good now thanks.